vault/resource_generic_secret_test.go

```diff
@@ -47,7 +47,33 @@ func TestResourceGenericSecret_deleted(t *testing.T) {
 					}
 				},
 				Config: testResourceGenericSecret_initialConfig(path),
-				Check:  testResourceGenericSecret_initialCheck(path),
+				Check:  testResourceGenericSecret_initialCheck(path, false),
+			},
+		},
+	})
+}
+
+func TestResourceGenericSecret_deleteKeyMetadata(t *testing.T) {
```

Comment:
May want to update the test name.

```suggestion
func TestResourceGenericSecret_deleteAllVersions(t *testing.T) {
```